dist/Devel-PPPort/parts/inc/call

```diff
@@ -54,7 +54,7 @@ __UNDEFINED__ PERL_LOADMOD_IMPORT_OPS   0x4
     (   (cond)                                      \
      && (_errsv = ERRSV)                            \
      && (SvROK(_errsv) || SvTRUE(_errsv))           \
-     && (croak_sv(_errsv), 1));                     \
+     && (({ croak_sv(_errsv); }), 1));              \
```

Comment:
This seems weird to me, wouldnt it be simpler to just define a croak_sv_ret() that returns 1? It would never actually do that as it would longjump, but wouldn't it be a simpler way to avoid the need for the list or the brace groups? It strikes me as something that is likely useful in other marcos.

Reply:
In core, `croak_sv` is a function. Devel::PPPort provides a version of it, but it's just a macro using `STMT_START`/`STMT_END`. That feels wrong. Maybe it could be changed like so:
```
PERL_STATIC_INLINE void D_PPP_CROAK_SV(pTHX_ SV *sv) {
    SV *_sv = (sv);
    if (SvROK(_sv)) {
        sv_setsv(ERRSV, _sv);
        croak(NULL);
    } else {
        D_PPP_FIX_UTF8_ERRSV_FOR_SV(_sv);
        croak("%" SVf, SVfARG(_sv));
    }
}
#  define croak_sv(sv) D_PPP_CROAK_SV(sv)
```

Reply:
My point is that I think the (croak_sv(),1) is there because Perl_croak_sv() does not return a value. It is not a true function, but rather a procedure with a void return. So you cant use it in an expression by itself like the one we are looking at. If we create a Perl_croak_sv_ret() which returns 0, or 1, or whatever (it doesnt matter it will never return). Then the compile will be happy and there would be no need for (croak_sv(),1).

Eg, if a function is void then i think 

```
int x = void_function();
```

is an error but

```
int x = (void_function(),1);
```
is not.

I think that the function you propose would have the same problem, even if it works around the issue with PPPort providing a macro form of it.

Reply:
Frankly it seems weird to me at a certain level that the croak() functions dont return something. It make them usable in expressions without hijinks, and the return can be casted away where we dont need it. I guess that boat has long sailed now tho.